ntoskrnl/fsrtl/fastio.c

```diff
@@ -1661,10 +1661,54 @@ FsRtlReleaseFile(IN PFILE_OBJECT FileObject)
     PFSRTL_COMMON_FCB_HEADER FcbHeader;
     PDEVICE_OBJECT DeviceObject;
     PFAST_IO_DISPATCH FastDispatch;
+    PEXTENDED_DRIVER_EXTENSION DriverExtension;
+    PFS_FILTER_CALLBACKS FilterCallbacks;
 
     /* Get Device Object and Fast Calls */
     FcbHeader = (PFSRTL_COMMON_FCB_HEADER)FileObject->FsContext;
     DeviceObject = IoGetRelatedDeviceObject(FileObject);
+
+    DriverExtension = (PEXTENDED_DRIVER_EXTENSION)DeviceObject->DriverObject->DriverExtension;
+    FilterCallbacks = DriverExtension->FsFilterCallbacks;
+
+    /* Check if Filter Callbacks are supported */
+    if (FilterCallbacks && FilterCallbacks->PreReleaseForSectionSynchronization)
+    {
+        NTSTATUS Status;
+        PVOID CompletionContext;
+
+        FS_FILTER_CALLBACK_DATA CbData;
+
+        RtlZeroMemory(&CbData, sizeof(CbData));
```

Comment:
I do think that this simplification should work as well. But should be tried to compile before accepting the proposal:
```suggestion
        FS_FILTER_CALLBACK_DATA CbData = { sizeof(CbData) };
```